packages/python/plotly/_plotly_utils/data_utils.py

```diff
@@ -0,0 +1,75 @@
+from io import BytesIO
+import base64
+from .png import Writer, from_array
+
+try:
+    from PIL import Image
+
+    pil_imported = True
+except ImportError:
+    pil_imported = False
+
+
+def array_to_imagestr(img, backend="pil", compression=4, ext="png"):
```

Comment:
the name here is maybe something to think about? technically what we are doing here is representing an image array as a [data URI](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs) so I would vote for something like `image_array_to_data_url` ?